### PR TITLE
community[patch]: Update root_validators ChatModels: ChatBaichuan, QianfanChatEndpoint, MiniMaxChat, ChatSparkLLM, ChatZhipuAI

### PR DESCRIPTION
**Pull request by maang-h**
This PR updates root validators for:

- ChatModels: ChatBaichuan, QianfanChatEndpoint, MiniMaxChat, ChatSparkLLM, ChatZhipuAI

Issues #22819

### CODE REVIEW

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         values["model_kwargs"] = extra
         return values
 
-    @root_validator()
+    @root_validator(pre=True)
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/baichuan.py:145]: :heavy_check_mark:

File: libs/community/langchain_community/chat_models/minimax.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ class Config:
 
         allow_population_by_field_name = True
 
-    @root_validator(allow_reuse=True)
+    @root_validator(pre=True, allow_reuse=True)
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/minimax.py:156]: :heavy_check_mark:

File: libs/community/langchain_community/chat_models/zhipuai.py [hunk 3]
```diff
@@ -261,7 +261,7 @@ class Config:
 
         allow_population_by_field_name = True
 
-    @root_validator()
+    @root_validator(pre=True)
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/zhipuai.py:264]: :heavy_check_mark:

File: libs/community/langchain_community/chat_models/sparkllm.py [hunk 4]
```diff
@@ -224,6 +224,11 @@ def validate_environment(cls, values: Dict) -> Dict:
             "IFLYTEK_SPARK_LLM_DOMAIN",
             SPARK_LLM_DOMAIN,
         )
+        return values
+
+    @root_validator(pre=False, skip_on_failure=True)
+    def post_init(cls, values: Dict) -> Dict:
+        """Post init validation for the class."""
         # put extra params into model_kwargs
         values["model_kwargs"]["temperature"] = values["temperature"] or cls.temperature
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/sparkllm.py:233]:
```suggestion
        values["model_kwargs"]["temperature"] = values["temperature"]
```


`cls.temperature` doesn't exist I think

File: libs/community/langchain_community/chat_models/sparkllm.py [hunk 5]
```diff
@@ -224,6 +224,11 @@ def validate_environment(cls, values: Dict) -> Dict:
...
         values["model_kwargs"]["top_k"] = values["top_k"] or cls.top_k
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/sparkllm.py:234]:
```suggestion
        values["model_kwargs"]["top_k"] = values["top_k"]
```

File: libs/community/langchain_community/chat_models/baidu_qianfan_endpoint.py [hunk 6]
```diff
@@ -189,6 +189,11 @@ def validate_environment(cls, values: Dict) -> Dict:
                 default="",
             )
         )
+        return values
+
+    @root_validator(pre=False, skip_on_failure=True)
+    def post_init(cls, values: Dict) -> Dict:
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_models/baidu_qianfan_endpoint.py:195]: This logic should be moved into the pre=True. It's not validation logic (the client should usually be instantiated before init), post_init is mainly for doing consistency checks that things are initialized properly

File: libs/community/langchain_community/chat_models/sparkllm.py (hunk 4 above)
Comment (maang-h) [reply libs/community/langchain_community/chat_models/sparkllm.py:233]: you are right. but same the below, it should  move back the pre=True? it didn't have any checks.

File: libs/community/langchain_community/chat_models/baidu_qianfan_endpoint.py (hunk 6 above)
Comment (maang-h) [reply libs/community/langchain_community/chat_models/baidu_qianfan_endpoint.py:195]: get.

**Comment by maang-h** [comment]
@eyurtsev cc, if this is ok, i will continue with other ChatModels and Embeddings.

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 7]
```diff
@@ -155,6 +155,7 @@ def validate_environment(cls, values: Dict) -> Dict:
                 values,
                 "baichuan_api_key",
                 "BAICHUAN_API_KEY",
+                values.get("api_key"),
```
[the remark this answers is not in the document]
Comment (maang-h) [reply libs/community/langchain_community/chat_models/baichuan.py]: get

**Pull request merged**